jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java

```diff
@@ -191,15 +194,35 @@ public void signalEvent(String type, Object event) {
         
         // process currently active runtime engines
         Map<Object, RuntimeEngine> currentlyActive = local.get();
+
         if (currentlyActive != null && !currentlyActive.isEmpty()) {
-            RuntimeEngine[] activeEngines = currentlyActive.values().toArray(new RuntimeEngine[currentlyActive.size()]);
-            for (RuntimeEngine engine : activeEngines) {
-                Context<?> context = ((RuntimeEngineImpl) engine).getContext();
+            @SuppressWarnings("unchecked")
+            Entry<Object, RuntimeEngine> activeEngines[] = currentlyActive.entrySet()
+                    .toArray(new Entry[currentlyActive.size()]);
```

Comment:
To avoid SuppressWarnings annotation, could it be like this?
```suggestion
            Map.Entry<Object, RuntimeEngine> activeEngines[] = currentlyActive.entrySet()
            .toArray(new Map.Entry[currentlyActive.size()]);
```

Reply:
I do not think it makes anny difference

Reply:
Hi, sorry for commenting a merged PR, I am not proposing any change. I just want to highlight that Gonzalo's comment actually makes a difference. If we let SuppressWarnings in the code like this, when there is a proposed solution that doesn't need it, the code will get polluted with the annotations. I know it doesn't make difference from functional perspective, however it makes sense from maintenance perspective. For future I would recommend to use the code that doesn't need this annotation.

Reply:
@baldimir i guess the suggested change was not having expected result - to not need to suppress the warning. it would be still required or see the compiler warning here.